### PR TITLE
Mgmt Container Instance migration and update

### PR DESCRIPTION
**Pull request by ChenTanyi**

### CODE REVIEW

File: sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java [hunk 1]
```diff
@@ -0,0 +1,1118 @@
...
+            /**
+             * Specifies the network profile information for a container group.
+             *
+             * @param subscriptionId the ID of the subscription of the network profile
+             * @param resourceGroupName the name of the resource group of the network profile
+             * @param networkProfileName the name of the network profile
+             * @return the next stage of the definition
+             */
+            DnsConfigFork withNetworkProfileId(
+                String subscriptionId, String resourceGroupName, String networkProfileName);
```
Comment (weidongxu-microsoft) [inline sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java:1024]: Should we provide a method with full id (if network profile got a full id)? And should we call it `withExistingNetworkProfile`?

File: sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java [hunk 2]
```diff
@@ -0,0 +1,1118 @@
...
+
+            /**
+             * Specifies the virtual network in network profile for a container group.
+             *
+             * @param virtualNetworkId the ID of the virtual network
+             * @param subnetName the name of the subnet within the virtual network.; the subnet must have the service
+             *     endpoints enabled for 'Microsoft.ContainerInstance/containerGroups'.
+             * @return the next stage of the definition
+             */
+            DnsConfigFork withExistingVirtualNetwork(String virtualNetworkId, String subnetName);
```
Comment (weidongxu-microsoft) [inline sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java:1034]: `withNewNetworkProfileOnExistingVirtualNetwork`, or `.withExistingVirtualNetwork(...).withNewNetworkProfile(...)`?

File: sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/implementation/ContainerGroupImpl.java [hunk 3]
```diff
@@ -0,0 +1,811 @@
...
+            this.volumeName = volumeName;
+            this.fileShareName = fileShareName;
+            this.storageAccountKey = storageAccountKey;
+        }
+    }
+
+    private Flux<VolumeParameters> createFileShareAsync(final StorageAccount storageAccount) {
+        return storageAccount
+            .getKeysAsync()
+            .map(storageAccountKeys -> storageAccountKeys.get(0).value())
```
Comment (weidongxu-microsoft) [inline sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/implementation/ContainerGroupImpl.java:167]: remember to run `credcheck` for storage key

File: sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java (hunk 1 above)
Comment (ChenTanyi) [reply sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java:1024]: This is a old function, I didn't change it.

File: sdk/management/azure/src/main/java/com/azure/resourcemanager/Azure.java [hunk 4]
```diff
@@ -682,13 +684,13 @@ public KubernetesClusters kubernetesClusters() {
...
-    //     * @return entry point to managing Azure Container Instances.
-    //     */
-    //    @Beta(SinceVersion.V1_3_0)
-    //    public ContainerGroups containerGroups() {
-    //        return containerInstanceManager.containerGroups();
-    //    }
+    /**
+     * @return entry point to managing Azure Container Instances.
+     */
+    @Beta
```
Comment (xccc-msft) [inline sdk/management/azure/src/main/java/com/azure/resourcemanager/Azure.java:690]: Why we add beta here?

Comment (ChenTanyi) [reply sdk/management/azure/src/main/java/com/azure/resourcemanager/Azure.java:690]: It is still a `beta` previously

File: sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/implementation/ContainerGroupImpl.java (hunk 3 above)
Comment (ChenTanyi) [reply sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/implementation/ContainerGroupImpl.java:167]: All playback tests are not using the `Share File Volume`, so no keys are recorded.

File: sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java]: If we want to do some breaking change, here is the best time. Anyway, up to you.

**Pull request merged**